prow/autobump.sh

```diff
@@ -53,8 +54,9 @@ ensure-config() {
 }
 ensure-config "$@"
 
-echo "Bumping prow to latest..." >&2
-./bump.sh --latest
+echo "Bumping images to latest..." >&2
+output=$(find "${TESTINFRA_ROOT}" -name '*.yaml' -exec bazel run //experiment/image-bumper -- --image-regex="gcr.io/(k8s-prow|k8s-testimages)/" "{}" +)
+body=$(echo "${output}" | grep -v gcr.io/k8s-prow | sort | sed -e 's/\(.*\):v\([0-9]\{8\}\)-\([0-9a-f]\+\)\(.*\) -> v\([0-9]\{8\}\)-\([0-9a-f]\+\).*/`\1\4`: https:\/\/github.com\/kubernetes\/test-infra\/compare\/\3...\6 (\2 to \5)/')
```

Comment:
What would be really cool is if we collect all the `{image: aaa...bbb}` transformations and then put them into a `{aaa...bbb: [image1, image2], ccc...ddd: [image3, image4]}` so that I can quickly review which images have which changes in them.

Maybe that's what this is doing? Regexes are kind of painful to read.

Reply:
Couple of comments:
- may be do a var as validImageTag=<regex>
- rewrite these two lines in a separate func so that we can have a more readable version.
- Add comments